### PR TITLE
Implement symfony validator

### PR DESCRIPTION
**Pull request by Soullivaneuh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,53 @@ final class MyEnum extends AbstractEnum
 }
 ```
 
+### Symfony validator
+
+This package provide a "ready to use" symfony validator.
```
Comment (greg0ire) [inline README.md]: provides

File: README.md [hunk 2]
```diff
@@ -120,6 +120,53 @@ final class MyEnum extends AbstractEnum
...
+You have to require `symfony/validator` package to get it working.
```
Comment (greg0ire) [inline README.md]: "require the"

File: README.md [hunk 3]
```diff
@@ -120,6 +120,53 @@ final class MyEnum extends AbstractEnum
...
+
+```php
+use Greg0ire\Enum\Bridge\Symfony\Validator\Constraint\Enum;
+use Symfony\Component\Validator\Validation;
+use Your\Namespace\EnumClass;
+
+$validator = Validation::createValidator();
+
+$violations = $validator->validateValue(42, new Enum(EnumClass::class));
+// You can also show the constants keys on the error message:
```
Comment (greg0ire) [inline README.md]: awesome!

File: src/Bridge/Symfony/Validator/Constraint/Enum.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($options = null)
+    {
+        parent::__construct($options);
+
+        if (!in_array('Greg0ire\Enum\AbstractEnum', class_parents($this->class))) {
+            throw new ConstraintDefinitionException('The option "enumClass" must be a class that inherit from Greg0ire\Enum\AbstractEnum');
```
Comment (greg0ire) [inline src/Bridge/Symfony/Validator/Constraint/Enum.php]: - "inherits"
- this line is too long

File: README.md (hunk 3 above)
Comment (Soullivaneuh) [inline README.md]: Is this a word you want to add to the readme? :trollface:

File: src/Bridge/Symfony/Validator/Constraint/Enum.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    public $showKeys = false;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($options = null)
+    {
+        parent::__construct($options);
+
+        if (!in_array('Greg0ire\Enum\AbstractEnum', class_parents($this->class))) {
```
Comment (greg0ire) [inline src/Bridge/Symfony/Validator/Constraint/Enum.php]: You could simply use `is_a`, maybe ?

File: test/Bridge/Symfony/Validator/Constraint/EnumTest.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        return array(
+            array('Greg0ire\Enum\Tests\Fixtures\AllEnum'),
+            array('Greg0ire\Enum\Tests\Fixtures\DummyEnum'),
+            array('Greg0ire\Enum\Tests\Fixtures\FooEnum'),
+        );
+    }
+
+    /**
+     * @dataProvider getInvalidEnum
+     * @expectedException \Symfony\Component\Validator\Exception\ConstraintDefinitionException
```
Comment (greg0ire) [inline test/Bridge/Symfony/Validator/Constraint/EnumTest.php]: Please use `setExpectedException`

File: test/Bridge/Symfony/Validator/Constraint/EnumTest.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public function testInvalidEnumClasses($enumClass)
+    {
+        new Enum($enumClass);
+    }
+
+    public function getInvalidEnum()
+    {
+        return array(
+            array('Greg0ire\Enum\Tests\Fixtures\FooInterface'),
```
Comment (greg0ire) [inline test/Bridge/Symfony/Validator/Constraint/EnumTest.php]: One element… are planning on adding new elements later?

Comment (Soullivaneuh) [inline test/Bridge/Symfony/Validator/Constraint/EnumTest.php]: Not really, but it's prepared for that. :-)

Do you want removing of the provider?

Comment (Soullivaneuh) [inline test/Bridge/Symfony/Validator/Constraint/EnumTest.php]: Oh I can add one! :-)

**Comment by Soullivaneuh** [comment]
@greg0ire Review fixed.

**Comment by greg0ire** [comment]
"commmit" => "commit"

**Comment by greg0ire** [comment]
"Symfony validator implement" => "implement symfony validator"

File: test/Bridge/Symfony/Validator/Constraint/EnumTest.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+        new Enum($enumClass);
+    }
+
+    public function getInvalidEnums()
+    {
+        return array(
+            array('Greg0ire\Enum\Tests\Fixtures\FooInterface'),
+            array('\StdClass'),
+            array('This\Not\Exists\At\All'),
```
Comment (greg0ire) [inline test/Bridge/Symfony/Validator/Constraint/EnumTest.php]: Does Not Exist

**Comment by Soullivaneuh** [comment]
@greg0ire all done. :+1:

**Pull request merged**